lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -607,79 +610,110 @@ Ufe::Attribute::Type usdTypeToUfe(const SdfValueTypeName& usdType)
+} // namespace
 
-Ufe::Attribute::Type usdTypeToUfe(const PXR_NS::SdrShaderPropertyConstPtr& shaderProperty)
+Ufe::Attribute::Type usdTypeToUfe(const SdrShaderPropertyConstPtr& shaderProperty)
 {
-    const PXR_NS::SdfValueTypeName typeName = shaderProperty->GetTypeAsSdfType().first;
-    if (typeName.GetHash() == PXR_NS::SdfValueTypeNames->Token.GetHash()) {
+    Ufe::Attribute::Type retVal = Ufe::Attribute::kInvalid;
+
+    const SdfValueTypeName typeName = shaderProperty->GetTypeAsSdfType().first;
+    if (typeName.GetHash() == SdfValueTypeNames->Token.GetHash()) {
         static const TokenToSdfTypeMap tokenTypeToSdfType
-            = { { PXR_NS::SdrPropertyTypes->Int, PXR_NS::SdfValueTypeNames->Int },
-                { PXR_NS::SdrPropertyTypes->String, PXR_NS::SdfValueTypeNames->String },
-                { PXR_NS::SdrPropertyTypes->Float, PXR_NS::SdfValueTypeNames->Float },
-                { PXR_NS::SdrPropertyTypes->Color, PXR_NS::SdfValueTypeNames->Color3f },
+            = { { SdrPropertyTypes->Int, SdfValueTypeNames->Int },
+                { SdrPropertyTypes->String, SdfValueTypeNames->String },
+                { SdrPropertyTypes->Float, SdfValueTypeNames->Float },
+                { SdrPropertyTypes->Color, SdfValueTypeNames->Color3f },
 #if defined(USD_HAS_COLOR4_SDR_SUPPORT)
-                { PXR_NS::SdrPropertyTypes->Color4, PXR_NS::SdfValueTypeNames->Color4f },
+                { SdrPropertyTypes->Color4, SdfValueTypeNames->Color4f },
 #endif
-                { PXR_NS::SdrPropertyTypes->Point, PXR_NS::SdfValueTypeNames->Point3f },
-                { PXR_NS::SdrPropertyTypes->Normal, PXR_NS::SdfValueTypeNames->Normal3f },
-                { PXR_NS::SdrPropertyTypes->Vector, PXR_NS::SdfValueTypeNames->Vector3f },
-                { PXR_NS::SdrPropertyTypes->Matrix, PXR_NS::SdfValueTypeNames->Matrix4d } };
+                { SdrPropertyTypes->Point, SdfValueTypeNames->Point3f },
+                { SdrPropertyTypes->Normal, SdfValueTypeNames->Normal3f },
+                { SdrPropertyTypes->Vector, SdfValueTypeNames->Vector3f },
+                { SdrPropertyTypes->Matrix, SdfValueTypeNames->Matrix4d } };
         TokenToSdfTypeMap::const_iterator it
             = tokenTypeToSdfType.find(shaderProperty->GetTypeAsSdfType().second);
         if (it != tokenTypeToSdfType.end()) {
-            Ufe::Attribute::Type type = usdTypeToUfe(it->second);
-            if (type == Ufe::Attribute::kString && !shaderProperty->GetOptions().empty()) {
-                return Ufe::Attribute::kEnumString;
-            } else {
-                return type;
-            }
+            retVal = _UsdTypeToUfe(it->second);
         } else {
 #if PXR_VERSION < 2205
             // Pre-22.05 boolean inputs are special:
             if (shaderProperty->GetType() == SdfValueTypeNames->Bool.GetAsToken()) {
-                return usdTypeToUfe(PXR_NS::SdfValueTypeNames->Bool);
-            }
+                retVal = _UsdTypeToUfe(SdfValueTypeNames->Bool);
```

Comment:
Isn't the `#if` logic wrong here? If the `#if` isn't defined won't the code be broken?

Reply:
No. The code becomes `if (Booltest) { kBool } else if (Matrix3dTest) { kMatrix3d } else { kGeneric }`

This is why there is a dangling `else` in the `#if` block.